### DISCUSSION TITLE
Mark different routes as "API routes"

### DISCUSSION THREAD

**Discussion by ivanpetrushev** [vercel/next.js#16817] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

We have to proxy several routes to a 3rd party service like in this example: https://github.com/vercel/next.js/discussions/14057#discussioncomment-31800 
This is working only in api routes and in Next.js api routes reside only in `pages/api/` directory.
We need paths other than `http://domain.com/api/` to be server as API routes.

## Describe the solution you'd like

Probably config list in next.config.js containing a list of directories in `pages/` that should be interpreted as API routes?
Probably HOC that can be used like:
```
import {withAsApiRoute} from "next-api-route";
...
export withAsApiRoute(Component);
```

I am open to other alternatives as well.

## Describe alternatives you've considered

Maybe we can use `rewrites` to change the API routes to desired paths, BUT - docs state rewrites are available only in Node.js environment and we are targeting serverless build.

Thanks.

**Comment by timneutkens** [comment]
Duplicate of https://github.com/vercel/next.js/discussions/13344
